server/src/test/java/io/druid/curator/discovery/ServiceAnnouncerTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.curator.discovery;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
+import com.metamx.common.ISE;
+import io.druid.curator.CuratorTestBase;
+import org.apache.curator.x.discovery.ServiceDiscovery;
+import org.apache.curator.x.discovery.ServiceDiscoveryBuilder;
+import org.apache.curator.x.discovery.ServiceInstance;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+public class ServiceAnnouncerTest extends CuratorTestBase
+{
+  @Before
+  public void setUp() throws Exception
+  {
+    setupServerAndCurator();
+  }
+
+  @Test
+  public void testServiceAnnouncement() throws Exception
+  {
+    curator.start();
+    List<String> serviceNames = ImmutableList.of(
+        "druid/overlord",
+        "druid/coordinator",
+        "druid/firehose/tranquility_test-50-0000-0000"
+    );
+    final ServiceDiscovery serviceDiscovery = createAndAnnounceServices(serviceNames);
+    Assert.assertTrue(
+        Iterators.all(
+            serviceNames.iterator(),
+            new Predicate<String>()
+            {
+              @Override
+              public boolean apply(String input)
+              {
+                try {
+                  return serviceDiscovery.queryForInstances(input.replaceAll("/", ":")).size() == 1;
+                }
+                catch (Exception e) {
+                  e.printStackTrace();
```

Comment:
Actually it can probably just be put in the ISE and let whatever catches it do the logging (JUnit probably)